### PR TITLE
Move some unit tests to outerloop to get clean CI

### PR DESCRIPTION
**Pull request by pgovind**
Just moving some tests to outerloop in the hope that it leads to clean CI and fixes https://github.com/dotnet/runtime/issues/41286. Relevant parts copied from https://github.com/dotnet/runtime/pull/2194

### CODE REVIEW

**Review by ericstj — APPROVED** [review]
LGTM assuming it succeeds.  These test both mention they OOM on 32-bit so it's a good chance they're causing the test to be killed on linux.

**Comment by pgovind** [comment]
@Anipik: You said something about checking if the branch is open. Not sure exactly what you meant by that, so I'll leave it to you to merge this. Whenever you get a chance, this PR is good to "Squash and merge"

**Review by Anipik — APPROVED** [review]

**Pull request merged**
